src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSTags.java

```diff
@@ -0,0 +1,12 @@
+package by.dragonsurvivalteam.dragonsurvival.registry;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.ItemTags;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.item.Item;
+
+import static by.dragonsurvivalteam.dragonsurvival.DragonSurvivalMod.MODID;
+
+public class DSTags {
+    public static final TagKey<Item> VAULT_KEYS = ItemTags.create(ResourceLocation.fromNamespaceAndPath(MODID, "vault_keys"));
+}
```

Comment:
https://github.com/DragonSurvivalTeam/DragonSurvival/blob/1.21.x/src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/datagen/DSItemTags.java?

Reply:
why not use the existing class

Reply:
I removed this class and opted to store the relevant data in a custom data component attached to the key instead.

Reply:
That way, instead of only having a single TagKey that opens a single vault type, you can have arbitrarily many tags to open arbitrarily many vaults with arbitrarily many possible keys